BEProjectsApp/models.py

```diff
@@ -80,3 +98,50 @@ def publish(self):
 
     def __str__(self):
         return self.title
+
+
+class OH_Project(models.Model):
+
+    # Store the Project Title
+    name = models.CharField(max_length=100, blank=False, default="Untitled Project")
+    # Store the ID of the Teacher model who is mentoring the project
+    teacher_id = models.ForeignKey(
+        Teacher, related_name="teacher", on_delete=models.CASCADE
+    )
+    # project description
+    description = models.TextField()
+    # year published and created will be stored
+    year_created = models.DateTimeField(default=timezone.now)
+    year_published = models.DateTimeField(blank=True, null=True)
+    # to store the files uploaded
+    document = models.FileField()
+    # domain list
+    domain = models.CharField(
+        choices=DOMAIN_CHOICES_OHP, default="none", blank=False, max_length=100
+    )
+    # out house details
+    company = models.CharField(max_length=100, blank=False, default="none")
+    supervisor = models.CharField(max_length=100, blank=False, default="none")
+    # contributers stored as a text field
+    contributers = models.TextField()
+
+    """
+
+    The purpose of the pubish function is to create the functionality of drafts where
+
+    the teachers get to approve a project and then it will be published till then
+
+    only created date is stored.
+
+    The display of search results will only be done with the published date so if that
+
+    is empty it should not be shown in the search results.
+
+    """
+
+    def publish(self):
+        self.published_date = timezone.now()
+        self.save()
+
+    def __str__(self):
+        return self.name
```

Comment:
You missed the `approved` attribute that is a Boolean field. Because we need to showcase only teacher verified projects. And you have missed because we also forgot to write it in the schema design.

Reply:
for that i added this publish feature so created date and published date so that's what I thought to do in the first place

Reply:
i'll change it to approve

Reply:
Your `publish` function is good but we need an attribute `approve` separately so if it becomes true we can call `obj.publish()`.

Reply:
check the new commit

Reply:
now there will be no need of publish also  
if approved then include in the query set
else no